### PR TITLE
Fixed argument spec for multiple modules - Part II

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

This change contains fixes for argument spec and respective datatypes.
Created separate PR since these changes might alter the behavior of these modules.
Will need shipit from individual module owner(s).

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE 
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/spec_fix.yml
lib/ansible/modules/cloud/google/gce_instance_template.py
lib/ansible/modules/cloud/openstack/os_subnet.py
lib/ansible/modules/cloud/openstack/os_zone.py
lib/ansible/modules/network/cnos/cnos_static_route.py
lib/ansible/modules/network/ios/ios_static_route.py
lib/ansible/modules/network/meraki/meraki_firewalled_services.py
lib/ansible/modules/network/meraki/meraki_nat.py
lib/ansible/modules/network/meraki/meraki_snmp.py
lib/ansible/modules/network/meraki/meraki_ssid.py
lib/ansible/modules/network/meraki/meraki_syslog.py

### CODE REVIEW

**Review by Qalthos — APPROVED** [review]

File: lib/ansible/modules/network/meraki/meraki_snmp.py [hunk 1]
```diff
@@ -317,7 +317,7 @@ def main():
                          peer_ips=dict(type='str'),
                          access=dict(type='str', choices=['none', 'community', 'users']),
                          community_string=dict(type='str', no_log=True),
-                         users=dict(type='list', default=None, element='str', options=user_arg_spec),
+                         users=dict(type='list', default=None, elements='str', options=user_arg_spec),
```
Comment (felixfontein) [inline lib/ansible/modules/network/meraki/meraki_snmp.py]: If `elements='str'`, `options` doesn't make sense. I assume it should have been `elements='dict'` right from the beginning? @kbreit can you take a look?

**Review by felixfontein — COMMENTED** [review]
Changes look OK superficially (except the one thing below), as you said the module authors/maintainers have to take a closer look.

The changes to `cnos_static_route` and `ios_static_route` are safe, since `type='str'` is the default value when `type` is not specified.

**Review by felixfontein — APPROVED** [review]
I looked at all changes, I think they should be ok. The Meraki modules probably have the changes with the largest possibility of being problematic, in case the sub-option description is incorrect somewhere. Assuming it is correct, errors should now be detected before passing data on to the API.

**Comment by felixfontein** [comment]
(I've pinged @kbreit via IRC for the Meraki modules.)

**Comment by kbreit** [comment]
Trying to test but getting these errors. I already have template in place though.

```
(venv) ansible [spec_fix_lethal●] % ansible-test network-integration --allow-unsupported --docker default --python 3.7 meraki_firewalled_services meraki_nat meraki_snmp meraki_ssid meraki_syslog
ERROR: Inventory not found: /root/ansible/test/integration/inventory.networking
Use --inventory to specify the inventory path.
Use --platform to provision resources and generate an inventory file.
See also inventory template: /root/ansible/test/lib/ansible_test/config/inventory.networking.template
```

**Comment by kbreit** [comment]
I tested this in my collection and it works, but I had to make a few changes. What's the best way for me to submit the changes to this PR?

**Comment by Akasurde** [comment]
> I tested this in my collection and it works, but I had to make a few changes. What's the best way for me to submit the changes to this PR?


You can comment in this PR or itself, I will encorporate those changes in.

**Comment by kbreit** [comment]
@Akasurde I created a diff but there are other changes in it so it isn't totally clean. If this isn't the best way to do this, let me know. I'd do the comments but it didn't seem to be as complete.

If you know why my `ansible-test` isn't working, let me know and I can work off your branch directly and do full testing. I know this process isn't best practice.

https://paste.ee/p/JyaHq

Here are my notes
- The import statements are part of another cleanup so there isn't a need to integrate them. In fact, being too literal will bring the collection import statement in, which won't work.
- There's a change on line 24 and 25 which fixes another bug. If you want to move that to another PR, let me know.
- These changes made me need to update the documentation.
- Removing the result assignment and a payload assignment was in another cleanup.

**Comment by sshnaidm** [comment]
Thanks for submitting patch for Openstack Ansible modules!
We moved Openstack Ansible modules to Openstack repositories.
Next patches should be submitted not with Ansible Github but with
Openstack Gerrit: https://review.opendev.org/#/q/project:openstack/ansible-collections-openstack
Please submit your code there from now.
Thanks for your contribution and sorry for inconvienience.

**Comment by sshnaidm** [comment]
@Akasurde changes to openstack files won't be included in 2.10, see my message above:

> Thanks for submitting patch for Openstack Ansible modules!
We moved Openstack Ansible modules to Openstack repositories.
Next patches should be submitted not with Ansible Github but with
Openstack Gerrit: https://review.opendev.org/#/q/project:openstack/ansible-collections-openstack
Please submit your code there from now.
Thanks for your contribution and sorry for inconvienience.

**Comment by Akasurde** [comment]
@sshnaidm I created git review - https://review.opendev.org/#/c/703554/

**Comment by kbreit** [comment]
@Akasurde Will this be backported to 2.9? My modules will be migrated out to collections in 2.10 so I'm debating if I should backport.

**Comment by felixfontein** [comment]
I think it makes sense to backport this, especially for all docs-related changes.

**Pull request merged**